File: ui-preview/src/main/java/l/files/ui/preview/Cache.java
package l.files.ui.preview;

import android.support.annotation.Nullable;

import java.io.IOException;

import l.files.fs.Path;
import l.files.fs.Stat;
import l.files.ui.base.graphics.Rect;

interface Cache<V> {

    @Nullable
    V get(Path path, Stat stat, Rect constraint, boolean matchTime)
            throws IOException;

    @Nullable
    Snapshot<V> put(Path path, Stat stat, Rect constraint, V value)
            throws IOException;

}

File: ui-preview/src/main/java/l/files/ui/preview/Cache.kt
package l.files.ui.preview

import l.files.ui.base.graphics.Rect
import java.io.IOException
import java.nio.file.Path
import java.time.Instant

internal data class Snapshot<V>(
  val value: V,
  val time: Instant
)

internal interface Cache<V> {

  @Throws(IOException::class)
  fun get(
    path: Path,
    time: Instant,
    constraint: Rect,
    matchTime: Boolean
  ): V?

  @Throws(IOException::class)
  fun put(
    path: Path,
    time: Instant,
    constraint: Rect,
    value: V
  ): Snapshot<V>?
}
